### ISSUE TITLE
CheckBoxPreference misses Lollipop style click animation

### ISSUE THREAD

**Issue by jenzz** [jenzz/Android-MaterialPreference#3]
Title says it all...

**Comment by jenzz** [comment]
**Note**: The official [SwitchCompat](https://developer.android.com/reference/android/support/v7/widget/SwitchCompat.html) widget doesn't animate either...

**Comment by msutherland4807** [comment]
I've found a good workaround for these issues is to create a `xml-v21` with XML that mirrors your preferences with their native counterparts

**Comment by jenzz** [comment]
Sure, @msutherland4807. However, that would enable animations on Lollipop only, so it's not really backward-compatible.
Also, I personally don't like to maintain two almost identical `preferences.xml` files. My idea of this little library was to have one XML file that provides consistent Material design widgets across all API levels.

I'm still hoping for an update to the AppCompat library which backports some of these things and also moves widgets like `TintCheckBox` out of the _internal_ package into public scope.

**Comment by msutherland4807** [comment]
Agreed entirely. I also just noticed that the new Nova Launcher material beta has animations on it's Switches (but not CheckBoxes), perhaps that could help with this. I'll look into the apk when I get a chance

**Comment by Relaxuma** [comment]
The problem should be caused by the missing of the following code.


    private class Listener implements CompoundButton.OnCheckedChangeListener {
        @Override
        public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
            if (!callChangeListener(isChecked)) {
                // Listener didn't like it, change it back.
                // CompoundButton will make sure we don't recurse.
                buttonView.setChecked(!isChecked);
                return;
            }

            SwitchPreference.this.setChecked(isChecked);
        }
    }

and the following code need to be added in bind view as well

    switchView.setOnCheckedChangeListener(mListener);
